sound/soc/sof/sof-client.c

```diff
@@ -624,3 +624,9 @@ enum sof_fw_state sof_client_get_fw_state(struct sof_client_dev *cdev)
 	return sdev->fw_state;
 }
 EXPORT_SYMBOL_NS_GPL(sof_client_get_fw_state, "SND_SOC_SOF_CLIENT");
+
+struct snd_sof_dev *sof_client_dev_to_sof_dev(struct sof_client_dev *cdev)
+{
+	return cdev->sdev;
+}
+EXPORT_SYMBOL_NS_GPL(sof_client_dev_to_sof_dev, "SND_SOC_SOF_CLIENT");
```

Comment:
@ujfalusi I thought that client devices shouldn't deal with `struct snd_sof_dev` at all, then why is this function exported?

Reply:
@lyakh, right, the other core used functions also have been exported with this symbol, the other option is to not use namespace for these at all. The definition is in sof-priv.h, which is also forbidden to be included by client drivers.

We can make this over complicated, but I think during review this can be spotted and enforced.